### ISSUE TITLE
Add multi-valued field support to DistinctCountHLL

### ISSUE THREAD

**Issue by kkrugler** [apache/incubator-pinot#6642]
Currently the code in `DistinctCountHLLAggregationFunction.aggregate()` looks like:

``` java
  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
      Map<ExpressionContext, BlockValSet> blockValSetMap) {
    BlockValSet blockValSet = blockValSetMap.get(_expression);
    DataType valueType = blockValSet.getValueType();

    if (valueType != DataType.BYTES) {
      HyperLogLog hyperLogLog = getDefaultHyperLogLog(aggregationResultHolder);
      switch (valueType) {
        case INT:
          int[] intValues = blockValSet.getIntValuesSV();
          for (int i = 0; i < length; i++) {
            hyperLogLog.offer(intValues[i]);
          }
          break;
        case LONG:
         ...
```

Note the call to `blockValueSet.getIntValuesSV()`. It should check if `BlockValueSet.isSingleValue()` returns false, and if so then use `BlockValueSet.getIntValuesMV()`. This will return a two dimensional array (`[][]`) of values that need iteration.

The same change is needed for all of the other value types.

**Comment by Jackie-Jiang** [comment]
Currently, in order to aggregate on multi-valued fields, you need to use the MV version of the function: `DistinctCountHLLMVAggregationFunction`.
In the future, we might merge these 2 versions to make it easier to use.

**Comment by kkrugler** [comment]
Hi @Jackie-Jiang Updated issue to cover merging the two.

**Comment by kkrugler** [comment]
I imagine similar merging could happen for many of the other [xxxMV](https://docs.pinot.apache.org/users/user-guide-query/supported-aggregations#multi-value-column-functions) multi-value column functions.

**Comment by fx19880617** [comment]
right, we should be able to infer the type with the table schema during query parsing phase and then rewrite the query if it's a multi-value column.

**Comment by Jackie-Jiang** [comment]
@fx19880617 We don't really need to rewrite. We have access to the column metadata during aggregation, and knows whether it is SV or MV

**Comment by fx19880617** [comment]
Right, from the implementation side, we will eventually merge the behavior of `<function>` and `<function>MV`. Then we can rewrite all `<function>MV` to `<function>`.
